packages/compat/src/resolver-transform.ts

```diff
@@ -167,7 +175,11 @@ class TemplateResolver implements ASTPlugin {
 
   private moduleResolver: Resolver;
 
-  constructor(private env: Env, private config: CompatResolverOptions) {
+  constructor(
+    private env: Env,
+    private config: CompatResolverOptions,
+    private builtInsForEmberVersion: ReturnType<typeof builtInKeywords>
```

Comment:
since you're converting it to a function, why would you pass the function to the constructor and not just make it part of the class? 🤔

Reply:
not passing the function to the class here, just the result of the function invocation
as it needs the result and don't want to invoke the function inside the class due to perf reasons of not evaluating the semver more than once

Reply:
you don't need memoize as it's only called once at the place I call it, memoize would mean having to pass in the function and execute it inside each time to get back the same result
your ember version doesn't change (hopefully during the execution of the build) so this would be fewer function invocations that need to look up memoized content

Reply:
all seems good to me 😂 thanks for humouring me and my questions 🎉

Reply:
that's what the process is for :)